exasol_advanced_analytics_framework/udf_communication/frontend_peer_state.py

```diff
@@ -0,0 +1,64 @@
+import contextlib
+import time
+from typing import Optional, Generator, List
+
+import zmq
+
+from exasol_advanced_analytics_framework.udf_communication.connection_info import ConnectionInfo
+from exasol_advanced_analytics_framework.udf_communication.get_peer_receive_socket_name import \
+    get_peer_receive_socket_name
+from exasol_advanced_analytics_framework.udf_communication.messages import Message, WeAreReadyToReceiveMessage, \
+    AreYouReadyToReceiveMessage, PayloadMessage, PeerIsReadyToReceiveMessage
+from exasol_advanced_analytics_framework.udf_communication.peer import Peer
+from exasol_advanced_analytics_framework.udf_communication.serialization import serialize_message
+
+
+class FrontendPeerState:
+
+    def __init__(self,
+                 my_connection_info: ConnectionInfo,
+                 context: zmq.Context,
+                 peer: Peer):
+        self._my_connection_info = my_connection_info
+        self._peer = peer
+        self._context = context
+        self._peer_is_ready = False
+        self._create_receive_socket()
+
+    def _create_receive_socket(self):
+        self._receive_socket = self._context.socket(zmq.PAIR)
```

Comment:
same here, the name is a bit confusing. maybe rename to "backend_data_socket"

Reply:
I would argue that receive socket is here actually better, because its only purpose is that we receive over it payload messages from the specific peer, that the backend forwards them to us is a detail

Reply:
Ok, that's fine.